site/testsuites/testsuite-javascript/__tests__/tbdex/issuer/known customer credential/knownCustomerCredentialIssuer.test.js

```diff
@@ -0,0 +1,499 @@
+          ],
+          grants: {
+            'urn:ietf:params:oauth:grant-type:pre-authorized_code':
+              generateUniquePreAuthCode(),
+          },
+        },
+        url: 'https://issuer.example.com/idv/form',
+      };
+    }
+    res.json(credentialOffer);
+  } catch (error) {
+    // Handle verification errors
+    return res.status(401).json({
+      errors: ['Invalid token', error.message],
+    });
+  }
+});
+// :snippet-end:
+
+// :snippet-start: wellKnownIssuerMetadataEndpointIssuerJs
+app.get('/.well-known/openid-credential-issuer', (req, res) => {
+  const issuerMetadata = {
+    credential_issuer: 'https://issuer.example.com',
+    credential_endpoint: 'https://issuer.example.com/credentials',
+    credential_configurations_supported: {
+      // type of credentials Issuer supports and what credential will look like
+      KnownCustomerCredential: {
+        format: 'jwt_vc_json',
+        scope: 'CustomerIdentity',
+        cryptographic_binding_methods_supported: ['did:example'],
+        credential_signing_alg_values_supported: ['ES256'],
+        credential_definition: {
+          type: ['VerifiableCredential', 'KnownCustomerCredential'],
+          credentialSubject: {
+            country: {
+              display: [{ name: 'Country', locale: 'en-US' }],
+            },
+          },
+        },
+        proof_types_supported: {
+          jwt: {
+            proof_signing_alg_values_supported: ['ES256'],
+          },
+        },
+        display: [
+          {
+            name: 'Known Customer Credential',
+            locale: 'en-US',
+            logo: {
+              url: 'https://issuer.example.com/public/logo.png',
+              alt_text: 'Issuer Logo',
+            },
+            background_color: '#FFFFFF',
+            text_color: '#000000',
+          },
+        ],
+      },
+    },
+  };
+
+  res.json(issuerMetadata);
+});
+// :snippet-end:
+
+// :snippet-start: wellKnownAuthorizationServerEndpointIssuerJs
+app.get('/.well-known/oauth-authorization-server', (req, res) => {
+  const oauthAuthorizationServerMetadata = {
+    issuer: 'https://issuer.example.com', // URL of the Credential Issuer
+    token_endpoint: 'https://issuer.example.com/token', // URL for the Access Token Request
+  };
+
+  res.json(oauthAuthorizationServerMetadata);
+});
+// :snippet-end:
+
+// :snippet-start: accessTokenEndpointIssuerJs
+import { VerifiableCredential, Jwt } from '@web5/credentials';
+
+const accessTokenToCNonceMap = new Map();
+
+app.post('/token', async (req, res) => {
+  const { grant_type, code } = req.body;
+
+  if (grant_type !== 'urn:ietf:params:oauth:grant-type:pre-authorized_code') {
+    return res.status(400).json({ error: 'unsupported_grant_type' });
+  }
+
+  const customersDidUri = preAuthCodeToDidMap.get(code);
+  if (!customersDidUri) {
+    return res.status(400).json({ error: 'invalid_grant' });
+  }
+
+  /*******************************************
```

Comment:
Part 1 of my proposal for how we introduce the `authorization_pending` into this guide

```suggestion
  }

  const idvCompleted = checkIDVStatus(customersDidUri); // some function to check the status of the IDV
  if (idvCompleted) {
    return res.status(400).json({ error: 'authorization_pending' });
  }

  /*******************************************
```